pkg/tools/etcd_helper.go

```diff
@@ -469,7 +471,16 @@ func (h *EtcdHelper) SetObj(key string, obj, out runtime.Object, ttl uint64) err
 
 // Pass an EtcdUpdateFunc to EtcdHelper.GuaranteedUpdate to make an etcd update that is guaranteed to succeed.
 // See the comment for GuaranteedUpdate for more detail.
-type EtcdUpdateFunc func(input runtime.Object) (output runtime.Object, ttl uint64, err error)
+type EtcdUpdateFunc func(input runtime.Object, node *etcd.Node, res *etcd.Response) (output runtime.Object, ttl *uint64, err error)
```

Comment:
The single struct reduces future refactoring work and keeps the interface intentional.  I'll swap to start there - I didn't like exposing the etcd client package into those classes either

> On May 23, 2015, at 12:24 PM, Daniel Smith <notifications@github.com> wrote:
> 
> In pkg/tools/etcd_helper.go:
> 
> > @@ -469,7 +471,16 @@ func (h *EtcdHelper) SetObj(key string, obj, out runtime.Object, ttl uint64) err
> >  
> >  // Pass an EtcdUpdateFunc to EtcdHelper.GuaranteedUpdate to make an etcd update that is guaranteed to succeed.
> >  // See the comment for GuaranteedUpdate for more detail.
> > -type EtcdUpdateFunc func(input runtime.Object) (output runtime.Object, ttl uint64, err error)
> > +type EtcdUpdateFunc func(input runtime.Object, node *etcd.Node, res *etcd.Response) (output runtime.Object, ttl *uint64, err error)
> I guess it doesn't matter that much, then.
> 
> —
> Reply to this email directly or view it on GitHub.
>